GoRogue/MapGeneration/Region.cs

```diff
@@ -9,17 +10,56 @@
 
 namespace GoRogue.MapGeneration
 {
+    /// <summary>
+    /// Event fired when a region's Area is changed.
+    /// </summary>
+    [PublicAPI]
+    public class RegionAreaChanged : EventArgs
+    {
+        /// <summary>
+        /// The region whose area is changing
+        /// </summary>
+        public readonly Region Item;
+
+        /// <summary>
+        /// The previous Area
+        /// </summary>
+        public readonly Area Old;
+
+        /// <summary>
+        /// The Area to which we update
+        /// </summary>
+        public readonly Area New;
```

Comment:
```suggestion
        public readonly PolygonArea NewValue;
```

Type and name shift to be more consistent with other parts of the library.  In particular, IMO the name `New` would be best to avoid, because it's close to `new` which is a keyword.